src/mon/MonCap.h

```diff
@@ -196,13 +196,22 @@ struct MonCap {
   }
 
   bool fs_name_capable(string_view fs_name, __u8 mask) {
-    for (auto& g: grants) {
+    for (auto& g : grants) {
       if (g.is_allow_all()) {
 	return true;
       }
-      if (g.fs_name.empty() || g.fs_name == fs_name) {
-	if (mask & g.allow) {
+
+      if ((g.fs_name.empty() || g.fs_name == fs_name) && (mask & g.allow)) {
 	  return true;
+      }
+
+      if (not g.profile.empty()) {
```

Comment:
```suggestion
      if (!g.profile.empty()) {
```

and IMO, no need to check for empty before iterating through `g.profile_grants`. otherwise, better off calling `expand_profile()` explicitly.

Reply:
Fixed, thanks!